ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java

```diff
@@ -71,6 +83,21 @@ public static ReactFontManager getInstance() {
     return typeface;
   }
 
+  /*
+   * This method allows you to load custom fonts from res/font folder as provided font family name.
+   * Fonts may be one of .ttf, .otf or XML (https://developer.android.com/guide/topics/ui/look-and-feel/fonts-in-xml).
+   * To support multiple font styles or weights, you must provide a font in XML format.
+   *
+   * ReactFontManager.getInstance().addCustomFont(this, "Srisakdi", R.font.srisakdi);
+   */
+  public void addCustomFont(Context context, @NonNull String fontFamily, int fontId) {
+    Typeface font = ResourcesCompat.getFont(context, fontId);
+    if (font != null) {
+      mCustomTypefaceCache.put(fontFamily, font);
+    }
+    mHasCustomTypeface = true;
```

Comment:
From @mdvacca:

doesn't this need to be inside the if (font != null) {
also, why not to just remove this variable and create a private method:

```
private boolean hasCustomTypeface()  {
    return mCustomTypefaceCache.isEmpty();
}
```
or maybe we don't even need this variable, it is just preventing a call to Map.containsKey?

Reply:
we pass resource id to this method, but that ID might not be a font, but anything. So font != null is making sure that we have a font with provided id.